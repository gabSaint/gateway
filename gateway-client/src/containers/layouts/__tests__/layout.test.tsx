import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import renderer from "react-test-renderer";

import Layout from "../layout";

describe("Layout", () => {
  let wrapperLayout: ShallowWrapper | ReactWrapper;
  let props: any;

  beforeEach(() => {
    props = {
      children: <p id="child" />,
    };
    wrapperLayout = shallow(<Layout {...props} />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Layout {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Navbar", () => {
    const navbar = wrapperLayout.find("Navbar");
    expect(navbar).toHaveLength(1);
  });

  it("renders flex container", () => {
    const flexcontainer = wrapperLayout.find("div#flex-container");
    expect(flexcontainer).toHaveLength(1);
  });

  it("renders children", () => {
    const children = wrapperLayout.find("p#child");
    expect(children).toHaveLength(1);
  });
});
