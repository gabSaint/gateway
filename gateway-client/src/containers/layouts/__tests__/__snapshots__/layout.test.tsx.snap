// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <div
    className="ui menu"
  >
    <div>
      Home
    </div>
    <div>
      Peripherals
    </div>
    <div
      className="right item"
    >
      <div
        className="ui icon input"
      >
        <input
          placeholder="Search Gateway..."
          type="text"
        />
        <i
          className="search icon"
        />
      </div>
    </div>
  </div>,
  <div
    className="ui container"
    id="flex-container"
  >
    <p
      id="children"
    />
  </div>,
]
`;
